crypto/poison_detector.go

```diff
@@ -78,22 +78,50 @@ func NewPoisonRecordKeyStoreWrapper(keyStore keystore.RecordProcessorKeyStore) k
 	}
 }
 
-// GetClientIDSymmetricKeys implementation of keystore.PrivateKeyStore with for poison records keys
+// GetClientIDSymmetricKeys implementation of keystore.PrivateKeyStore for poison records keys
 func (p PoisonRecordKeyStoreWrapper) GetClientIDSymmetricKeys([]byte) ([][]byte, error) {
 	return p.keyStore.GetPoisonSymmetricKeys()
 }
 
-// GetServerDecryptionPrivateKeys implementation of keystore.PrivateKeyStore with for poison records keys
+// GetClientIDEncryptionKey implementation of keystore.PrivateKeyStore for poison records symmetric key
+func (p PoisonRecordKeyStoreWrapper) GetClientIDEncryptionKey([]byte) ([]byte, error) {
+	encryptionKeys, err := p.keyStore.GetPoisonSymmetricKeys()
```

Comment:
will be great to fetch here 1 key too instead of array of keys

Reply:
This one seems kinda confusing for me. For filesystem keystores v1/v2 the methods were inside corresponding packages, I was able to use specific functions to get the key (one to read from file or one to read it from keyring). But here we're in different package that only interact with interfaces. Maybe missing something :c

Reply:
so we need extend keystore interface related to poison records too with similar function that returns 1 key